src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSearchValues.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.Lightup;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    /// <summary>
+    /// CA1870: <inheritdoc cref="UseSearchValuesTitle"/>
+    /// </summary>
+    public abstract class UseSearchValuesAnalyzer : DiagnosticAnalyzer
```

Comment:
Do we foresee a VB implementation for the analyzer/fixer in future? Otherwise, it would fine to move the entire code into the C# specific layer. I'll let you make a call here.

Reply:
I don't personally have any plans to add VB support here, but if someone from the community was looking to contribute it I don't see an issue.

Reply:
> but if someone from the community was looking to contribute it I don't see an issue.

Please file an issue for VB support, thanks!

Reply:
> Please file an issue for VB support, thanks!

Never mind, @stephentoub said `The API is span-based. It's possible to use from VB but the overall experience will be subpar`